apps/api/src/certificate/controllers/certificate.controller.ts

```diff
@@ -0,0 +1,18 @@
+import assert from "http-assert";
+import { singleton } from "tsyringe";
+
+import { Protected } from "@src/auth/services/auth.service";
+import { CreateCertificateResponse } from "../http-schemas/create-certificate.schema";
+import { CertificateService } from "../services/certificate.service";
+
+@singleton()
+export class CertificateController {
+  constructor(private readonly certificateService: CertificateService) {}
+
+  @Protected([{ action: "sign", subject: "UserWallet" }])
+  async create(): Promise<CreateCertificateResponse> {
+    const cert = await this.certificateService.create();
+    assert(cert, 404, "Certificate not found");
```

Comment:
I don't think this is to ever work. 404 on a requested resource for a create op is controversial a bit :)
anyways the called function will throw the proper error if the wallet not found
```suggestion
```

Reply:
True